src/components/DynamicFormContainer/DynamicFormLayout.tsx

```diff
@@ -1,53 +1,131 @@
+import Ajv from "ajv";
 import React, { FunctionComponent, useState } from "react";
 import { Redirect } from "react-router";
-import { useActiveFormContext } from "../../common/context/activeForm";
 import { useConfigContext } from "../../common/context/config";
-import { SvgIcon, SvgIconArrowLeft } from "../../UI/SvgIcon";
-import { Title } from "../../UI/Title";
+import { useFormsContext } from "../../common/context/forms";
+import { Button } from "../../UI/Button";
+import { SvgIcon, SvgIconTrash, SvgIconXCircle } from "../../UI/SvgIcon";
 import { ToggleSwitch } from "../../UI/ToggleSwitch";
 import { Container } from "../Container";
-import { ProgressBar } from "../ProgressBar";
+import { DeleteModal } from "./DeleteModal";
 import { DynamicForm } from "./DynamicForm";
+import { DynamicFormHeader } from "./DynamicFormHeader";
 
 export const DynamicFormLayout: FunctionComponent = () => {
   const { config } = useConfigContext();
+  const [showDeleteModal, setDeleteModal] = useState(false);
   const [isPreviewMode, setIsPreviewMode] = useState(false);
-  const { activeFormIndex, setActiveFormIndex } = useActiveFormContext();
-  if (activeFormIndex === undefined) return <Redirect to="/forms-selection" />;
-  const activeForm = config?.forms[activeFormIndex];
-  if (!activeForm) return <Redirect to="/forms-selection" />;
+  const {
+    forms,
+    setForms,
+    setActiveFormIndex,
+    currentForm,
+    setCurrentFormData,
+  } = useFormsContext();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [formError, setFormError] = useState<Ajv.ErrorObject[] | null | undefined>();
+  if (!currentForm) return <Redirect to="/forms-selection" />;
+  const currentFormDefinition = config?.forms[currentForm?.templateIndex];
+  if (!currentFormDefinition) return <Redirect to="/forms-selection" />;
+  if (isSubmitted) return <Redirect to="/publish" />;
 
-  const onBackToFormSelection = (): void => {
+  const formSchema = currentFormDefinition.schema;
+  const attachmentAccepted = !!currentFormDefinition.attachments?.allow;
+  const attachmentAcceptedFormat = currentFormDefinition.attachments?.accept;
+
+  const validateCurrentForm = (): boolean => {
+    const ajv = new Ajv();
+    let hasError = false;
+    forms.forEach((form) => {
+      const validForm = ajv.validate(form.data.schema, form.data.formData);
+      if (!validForm) {
+        console.error(ajv.errors);
+        setFormError(ajv.errors);
+        return (hasError = true);
+      }
+      setFormError(ajv.errors);
+      return (hasError = false);
```

Comment:
Consider a loop that runs twice.
The first run, return errors
The second run, returns no errors
Should the final hasError be true or false?
In this case it seeemed like if the last loop has no error, then it will return no error.

Reply:
hmmm ok will edit it to cater to the error in any run of the loop